### PR TITLE
[Improvement]: Add TableFormat information to the return value of the Catalog listTable method.

### PR DESCRIPTION
**Pull request by baiyangtx**
## Why are the changes needed?

This PR is part of this issue #1061 . 

In order for a catalog to support multiple formats, we need to include `TableFormat` information in the return value of the catalog's listTable interface. The code changes related to the #1061  are very large. To reduce the burden of review, some code refactoring has been proposed as a separate PR.

## Brief change log
 
- Change the return type of `listTable` from `List<TableIdentifier>` to `List<TableIDWithFormat>`
- Add `format` to `ServerTableIdentifier`
- Remove `format` from `table_runtime`
- Change the input parameter of `ServerTableDescriptor` from `ServerTableIdentifier` to `TableIdentifier`.

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (yes / no)
- If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: ams/server/src/main/java/com/netease/arctic/server/table/TableManager.java [hunk 1]
```diff
@@ -1,10 +1,11 @@
 package com.netease.arctic.server.table;
 
 import com.netease.arctic.AmoroTable;
+import com.netease.arctic.ams.api.TableIdentifier;
 
 public interface TableManager extends TableRuntimeHandler {
 
-  AmoroTable<?> loadTable(ServerTableIdentifier tableIdentifier);
+  AmoroTable<?> loadTable(TableIdentifier tableIdentifier);
```
Comment (zhoujinsong) [inline ams/server/src/main/java/com/netease/arctic/server/table/TableManager.java]: I suggest keeping `ServerTableIdentifier` as the parameter to maintain that this interface is only for managed tables.

File: core/src/test/java/com/netease/arctic/formats/AmoroCatalogTestHelper.java [hunk 2]
```diff
@@ -109,4 +110,6 @@ public interface AmoroCatalogTestHelper<T> {
    */
   void createTable(String db, String tableName)
       throws Exception;
+
+  TableFormat format();
```
Comment (shidayang) [inline core/src/test/java/com/netease/arctic/formats/AmoroCatalogTestHelper.java:114]: Why need this?

File: ams/server/src/main/java/com/netease/arctic/server/table/TableManager.java (hunk 1 above)
Comment (baiyangtx) [reply ams/server/src/main/java/com/netease/arctic/server/table/TableManager.java]: fixed.

File: core/src/main/java/com/netease/arctic/UnifiedCatalog.java [hunk 3]
```diff
@@ -36,7 +36,7 @@ public interface UnifiedCatalog extends AmoroCatalog {
    * @param database given database
    * @return identifier and format list
    */
-  List<TableMeta> listTableMetas(String database);
+  List<TableIDWithFormat> listTableMetas(String database);
```
[the remark this answers is not in the document]
Comment (baiyangtx) [reply core/src/main/java/com/netease/arctic/UnifiedCatalog.java]: fixed.

File: core/src/test/java/com/netease/arctic/formats/AmoroCatalogTestHelper.java (hunk 2 above)
Comment (baiyangtx) [reply core/src/test/java/com/netease/arctic/formats/AmoroCatalogTestHelper.java]: fixed.

File: ams/server/src/main/java/com/netease/arctic/server/table/ServerTableIdentifier.java [hunk 4]
```diff
@@ -10,28 +11,32 @@ public class ServerTableIdentifier {
   private String catalog;
   private String database;
   private String tableName;
+  private TableFormat format;
```
Comment (wangtaohz) [inline ams/server/src/main/java/com/netease/arctic/server/table/ServerTableIdentifier.java:14]: After adding the field `format` to `ServerTableIdentifier`, we can remove this field from `TableMetadata`.
```java
public class TableMetadata implements Serializable {
  ...

  private ServerTableIdentifier tableIdentifier;

  private TableFormat format;
  ...
}
```

**Review by wangtaohz — COMMENTED** [review]
I left some comments, PTAL @baiyangtx

File: ams/server/src/main/java/com/netease/arctic/server/persistence/mapper/TableMetaMapper.java [hunk 5]
```diff
@@ -213,7 +216,7 @@ Integer deleteTableIdByName(
       @Param("databaseName") String databaseName,
       @Param("tableName") String tableName);
 
-  @Select("SELECT table_id, catalog_name, db_name, table_name FROM table_identifier" +
+  @Select("SELECT table_id, catalog_name, db_name, table_name, format FROM table_identifier" +
       " WHERE catalog_name = #{catalogName} AND db_name = #{databaseName} AND table_name = #{tableName}")
   @Results({
       @Result(property = "id", column = "table_id"),
```
[the remark this answers is not in the document]
Comment (baiyangtx) [reply ams/server/src/main/java/com/netease/arctic/server/persistence/mapper/TableMetaMapper.java:223]: fixed.

File: ams/server/src/main/java/com/netease/arctic/server/table/ServerTableIdentifier.java (hunk 4 above)
Comment (baiyangtx) [reply ams/server/src/main/java/com/netease/arctic/server/table/ServerTableIdentifier.java:14]: fixed.

File: ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/TableController.java [hunk 6]
```diff
@@ -302,15 +299,15 @@ public void getTableTransactions(Context ctx) {
    * @param ctx - context for handling the request and response
    */
   public void getTransactionDetail(Context ctx) {
-    String catalogName = ctx.pathParam("catalog");
-    String db = ctx.pathParam("db");
-    String tableName = ctx.pathParam("table");
+    String catalog = ctx.pathParam("catalog");
+    String database = ctx.pathParam("db");
+    String tableMame = ctx.pathParam("table");
```
[the remark this answers is not in the document]
Comment (baiyangtx) [reply ams/server/src/main/java/com/netease/arctic/server/dashboard/controller/TableController.java]: fixed.

**Review by wangtaohz — APPROVED** [review]
LGTM

**Review by shidayang — APPROVED** [review]

**Comment by shidayang** [comment]
LGTM

**Pull request merged**